src/psyclone/tests/test_files/gocean1p0/boundary_conditions_ne_offset_mod.f90

```diff
@@ -16,10 +16,10 @@ module boundary_conditions_ne_offset_mod
   !=======================================
 
   type, extends(kernel_type) :: bc_ssh
-     type(go_arg), dimension(3) :: meta_args =                 &
-          (/ go_arg(GO_READ,      GO_I_SCALAR, GO_POINTWISE),  &
-             go_arg(GO_READWRITE, GO_CT,       GO_POINTWISE),  &
-             go_arg(GO_READ,                   GO_GRID_MASK_T) &
+     type(go_arg), dimension(3) :: meta_args =             &
+          (/ go_arg(GO_READ,  GO_I_SCALAR, GO_POINTWISE),  &
+             go_arg(GO_WRITE, GO_CT,       GO_POINTWISE),  &
```

Comment:
Not sure where these changes have come from but I don't think they're right - a boundary-condition kernel doesn't write to all elements of a field and therefore we must preserve those input values that are not modified.

Reply:
The code in question is:
```
    if(tmask(ji,jj) <= 0) return
    IF     (tmask(ji,jj-1) < 0) THEN
       ssha(ji,jj) = amp_tide * sin(omega_tide * rtime)
    ... similar else if tests with assignmne tto ssha()
```
The variable ``ssha`` is NOT read, so as far as I can see the declaration as 'readwrite' was wrong (and I just included this what I thought was a simple typo here). It led to problems when I tried to use these tests for gocean (lowering them from ``NemoLoopFuse`` to ``LoopFuse``). I think a completely unrelated test failed (since it tried to fuse something which then failed). If this is an issue, I am happy to revert this and handle this later.

Reply:
The problem here I think is the mapping between the access described in the metadata and the INTENT used in Fortran. We should probably give arguments that are `GH_WRITE` intent `INOUT`. Otherwise, if we just give it intent `OUT` then Fortran is well within its rights to wipe the array before passing it into the kernel. Obviously we don't want this to happen given that we will typically be only writing to part of the data array (even in a 'standard' kernel we don't write to the halos for instance). This is a change (I think) that @TeranIvy recently had to make for the LFRic API. If I've got all that right then I suggest creating a new issue for this and putting the meta-data and intent back to how it was along with a TODO.

Reply:
OK, I'll do that - but I believe the issue you are referring to is that if you pass an object to a subroutine, Fortran needs to read the object information (in order to get access to the pointer to the data it is writing). So I think the issue is that the meta-data should indicate "READ", but the intent must be "INOUT".

Reply:
#1215, but I believe the original issue (passing in objects, so read access is required) does not apply to gocean, which passes in Fortran arrays. But then I found several issues (in the same file) where a ``GO_READ`` is matched with ``intent(inout)``.

Reply:
Having slept and looked at this again, I think we should just change the kernel definition (I'd got confused and forgotten that we don't generate this) so that the field that is GO_WRITE has INTENT(inout). That should then keep everyone happy?
If you agree then please make this change and then we're good to go.

Reply:
`ssha` is only written and has `intent(inout)` so I'm happy.